articles/node-vs-python/index.md

```diff
@@ -0,0 +1,179 @@
+
+Due to the event loop architecture of Node.js, it is used for websites that involve a lot of concurrent processes. It is also suitable for Internet of Things solutions and projects.
+
+#### Frameworks and Expandability
+Python uses an interpreter called CPython that improves the performance of Python. There are many frameworks and libraries for full stack development.
+- [Django](https://www.djangoproject.com/) - A python web framework based on the Model-View-Controller Architecture
+- [Flask](https://flask.palletsprojects.com/en/1.1.x/) - A python micro-framework for web development
+- [Tensorflow](https://www.tensorflow.org/) - A python based Machine Learning platform
+
+Node.js has a ton of libraries that make it extensible, scalable and feature-rich.
+- [Express](https://expressjs.com/) - Unopinionated web framework for Node.js.
+- [Adonis](https://adonisjs.com/) - Full-Stack MVC Framework.
+- [Koa](https://koajs.com/) - Robust Foundation for Web Applications.
+
+#### Package Management
+Python uses a package manager called `pip`. It is extremely easy to use and has a ton of community-published packages. These packages can be found at the [Python Package Index](https://pypi.org/).
+
+It uses a file called **requirements.txt** to keep track of packages in a project.
+
+To install a package in python, just enter:
+
+```
+pip install <package-name>
+```
+
+Node.js uses a package manager called `npm`. The [npm registry](https://www.npmjs.com/) contains open source packages for Node.js, front-end web development, mobile apps and more.
+
+To keep track of packages in a project, Node.js uses a file called **package.json**
+
+To install a Node.js package, just type:
+
+```
+npm i <package-name>
+```
+
+### APIs in Node.js
+
+Let us dive into creating an API in Node.js our API has an endpoint _/read_ which reads data from a table, _sample_ in a MySQL Database.
+
+```js
+/* Importing Libraries */
+
+const mysql = require("mysql");
+const express = require("express");
+const bodyParser = require("body-parser");
+
+/* Initialize Express */
+
+const app = express();
+
+/* Body Parser */
+
+app.use(bodyParser.json());
+
+app.use(bodyParser.urlencoded({ extended: true }));
+
+/* MySQL Config */
+
+const connection = mysql.createConnection({
+  host: "<hostname>",
+  user: "<username>",
+  password: "<password>",
+  database: "<dbname>",
+});
+
+/* Initialize MySQL Connection */
+
+connection.connect((err) => {
+  if (err) throw err;
+  console.log("Successfully connected to the database.");
+});
+
+/* Define API Endpoint */
+
+app.get("/read", (req, res) => {
+  connection.query("SELECT * FROM sample", (err, res, fields) => {
+    if (err) throw err;
+    console.log(res);
+  });
+});
+```
+
+In the snippet above, we initialize a MySQL connection using the Node.js MySQL library. Whenever a get request is sent to the _/read_ endpoint, the database is read.
```

Comment:
Your explanation of the code is lacking. You should provide comments in the code that better explain what the various lines of code are for.